gateway/src/apicast/configuration_loader/remote_v2.lua

```diff
@@ -317,8 +317,10 @@ function _M:config(service, environment, version)
   if res.status == 200 then
     local proxy_config = cjson.decode(res.body).proxy_config
 
-    local config_service = configuration.parse_service(proxy_config.content)
-    local issuer, oidc_config = self:oidc_issuer_configuration(config_service)
+    local oidc_issuer_endpoint = proxy_config.content.proxy and
```

Comment:
Note: when calling `parse_service`, the `configuration` module builds the policy chain, and when doing so, in the case of the headers policy, the configuration received in `new()` is modified because it adds a `template_string` field with an instance of `TemplateString`.

That's why this failed only when the service was parsed. Parsing the service is not a problem by itself, but later, when the proxy configs are encoded, it fails because of the template string.

An unexpected side-effect.

Reply:
We will need an integration test to reproduce this failure.

And I wonder why it crashes when we add an instance of TemplateString.

Reply:
I'll add an integration test.

Regarding the `TemplateString` exception, here's a minimal case to reproduce it:
```lua
local cjson = require 'cjson'
local Liquid = require 'liquid'
local LiquidTemplate = Liquid.Template

local template = LiquidTemplate:parse('{{ service.id }}')

cjson.encode(template) -- Error: Cannot serialise table: table key must be a number or string
```

Reply:
I see, the configuration loader tries to encode back the table into string:

https://github.com/3scale/apicast/blob/635590f8dedd895cbba4288faaecc13854967781/gateway/src/apicast/configuration_loader/remote_v2.lua#L128

Then we could do the deepclone just before parsing the service:
https://github.com/3scale/apicast/blob/635590f8dedd895cbba4288faaecc13854967781/gateway/src/apicast/configuration_loader/remote_v2.lua#L119

Reply:
That would work too :+1:

Reply:
At first I thought that there was no point in parsing the service, because the information can be easily get from the JSON. But maybe parsing the service is better in case we're performing some validations there.